src/world.rs

```diff
@@ -174,17 +174,21 @@ impl<'a> Iterator for CreateEntities<'a> {
     }
 }
 
+
 trait StorageLock: Any + Send + Sync {
     fn del_slice(&self, &[Entity]);
 }
 
 mopafy!(StorageLock);
 
-impl<S: StorageBase + Any + Send + Sync> StorageLock for RwLock<S> {
+impl<T: Component> StorageLock for RwLock<MaskedStorage<T>> {
     fn del_slice(&self, entities: &[Entity]) {
+        use storage::PrivateStorage;
         let mut guard = self.write().unwrap();
         for &e in entities.iter() {
-            guard.del(e);
+            if guard.get_mask().contains(e.get_id()) {
```

Comment:
This is why the example is not working. This should be `remove()`, as right now this will leave the mask with a set bit that has been deleted.